src/ray/util/tests/pipe_logger_test.cc

```diff
@@ -41,28 +49,27 @@ TEST_P(PipeLoggerTest, NoPipeWrite) {
          absl::StrFormat("%d", pipe_buffer_size).data(),
          /*overwrite=*/1);
 
-  // TODO(core): We should have a better test util, which allows us to create a temporary
-  // testing directory.
+  // TODO(hjiang): We should have a better test util, which allows us to create a
+  // temporary testing directory.
   const std::string test_file_path = absl::StrFormat("%s.out", GenerateUUIDV4());
 
-  // Take the default option, which doesn't have rotation enabled.
-  StreamRedirectionOption logging_option{};
-  logging_option.file_path = test_file_path;
-  auto log_token = CreateRedirectionFileHandle(logging_option);
+  // Delete temporary file.
+  absl::Cleanup cleanup_test_file = [&test_file_path]() {
+    EXPECT_TRUE(std::filesystem::remove(test_file_path));
+  };
 
-  ASSERT_EQ(write(log_token.GetWriteHandle(), kLogLine1.data(), kLogLine1.length()),
-            kLogLine1.length());
-  ASSERT_EQ(write(log_token.GetWriteHandle(), kLogLine2.data(), kLogLine2.length()),
-            kLogLine2.length());
-  log_token.Close();
+  // Take the default option, which doesn't have rotation enabled.
+  StreamRedirectionOption stream_redirection_opt{};
+  stream_redirection_opt.file_path = test_file_path;
```

Comment:
This is what windows also supports after this PR, why is it a unix only test?

Reply:
You're absolutely correct, I wrote it that way for code simplicity; otherwise you will have part of the test case for linux only, another part for both windows / unix.